### PR TITLE
Tilesets with collection of images, closes #210

### LINKED ISSUE #210: Tiled: Support for tilesets made with collection of images
I want to be able to do something like this video using HaxeFlixel and Tiled. 

https://www.youtube.com/watch?v=yVgcFdHOl6w

I think that what needs to be improved is the support of tiles that are made of collection of images. If I try to use that kind of tilesets now, HaxeFlixel Crashes. 

I'm trying to make the necessary changes to support this. 

@gamedevsam, you made the initial implementation of tiled. Do you have any advised on how to approach this new feature?

### PR DESCRIPTION
**Pull request by Tembac**
Solves issue #210 

I checked the feedback by @Gama11 to my previous pull request. I'm also updating the tiled demo.

### CODE REVIEW

**Comment by Tembac** [comment]
I was missing to delete an old file.

File: flixel/addons/editors/tiled/TiledImageTile.hx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	public var id:String;
+	public var width:Float;
+	public var height:Float;
+	public var source:String;
+	
+	public function new(Source:Fast)
+	{
+		for (img in Source.nodes.image)
+		{
+			//id = Std.string(img.att.id);
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledImageTile.hx]: ?

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 2]
```diff
@@ -92,23 +74,125 @@ class TiledTileSet
...
+					
+					var id:Int = Std.parseInt(node.att.id);
+					tileImagesSources[id] = new TiledImageSet();
+					tileImagesSources[id].keys.set("id", Std.string(id));
+					tileImagesSources[id].extend(node);
 				}
+				
+				if (tileWidth > 0 && tileHeight > 0)
+				{
+					numRows = 0;//cast(imgWidth / tileWidth);
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: What's with the commented out logic here?

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 3]
```diff
@@ -92,23 +74,125 @@ class TiledTileSet
...
+					tileProps[id].keys.set("id", Std.string(id));
+					for (prop in node.nodes.properties)
+					{
+						tileProps[id].extend(prop);
+					}
+				}
+				
+				if (tileWidth > 0 && tileHeight > 0)
+				{
+					numRows = cast(imgWidth / tileWidth);
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Using `cast` for conversion to `Int` is unsafe / bad practice, use `Std.int()` instead.

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 4]
```diff
@@ -92,23 +74,125 @@ class TiledTileSet
...
+					margin = Std.parseInt(source.att.margin);
 				}
 				
-				var id:Int = Std.parseInt(node.att.id);
-				tileProps[id] = new TiledPropertySet();
-				tileProps[id].keys.set("id", Std.string(id));
-				for (prop in node.nodes.properties)
+				// read properties
+				properties = new TiledPropertySet();
+				for (prop in source.nodes.properties) {
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Flixel's codestyle has a newline before the `{`. Although with a single line like this, omitting the brackets is also allowed.

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 5]
```diff
@@ -92,23 +74,125 @@ class TiledTileSet
...
+					spacing = Std.parseInt(source.att.spacing);
+				}
+				if (source.has.margin) 
+				{
+					margin = Std.parseInt(source.att.margin);
+				}
+				
+				// read properties
+				properties = new TiledPropertySet();
+				for (prop in source.nodes.properties) {
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Flixel's codestyle has a newline before the `{`. Although with a single line like this, omitting the brackets is also allowed.

File: flixel/addons/editors/tiled/TiledTileSet.hx (hunk 2 above)
Comment (Tembac) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Sorry, I think something didn't uploaded properly. I though I changed this. I have trouble getting used to Github. 

Now I'm looking that I have the changes locally but they didn't uploaded.

Same for next comment.

**Comment by Tembac** [comment]
Updated files.

**Comment by Tembac** [comment]
Deleted useless line

**Pull request merged**